bot/exts/evergreen/fun.py

```diff
@@ -57,17 +57,17 @@ def __init__(self, bot: Bot) -> None:
         with Path("bot/resources/evergreen/caesar_info.json").open("r", encoding="UTF-8") as f:
             self._caesar_cipher_embed = json.load(f)
 
+    @staticmethod
+    def _get_random_die() -> str:
+        """Generate a random die emoji, ready to be sent on Discord."""
+        die_name = f"dice_{random.randint(1, 6)}"
+        return getattr(Emojis, die_name)
+
     @commands.command()
     async def roll(self, ctx: Context, num_rolls: int = 1) -> None:
         """Outputs a number of random dice emotes (up to 6)."""
-        def _get_random_die() -> str:
-            """Generate a random die emoji, ready to be sent on Discord."""
-            die_name = f"dice_{random.randint(1, 6)}"
-            return getattr(Emojis, die_name)
-
-        # Only support between 1 and 6 rolls
         if 1 <= num_rolls <= 6:
-            dice = " ".join(_get_random_die() for _ in range(num_rolls))
+            dice = " ".join(Fun._get_random_die() for _ in range(num_rolls))
```

Comment:
```suggestion
            dice = " ".join(self._get_random_die() for _ in range(num_rolls))
```
It's fine that `_get_random_die` is a `@staticmethod`, but we should still call it with `self`. Using the class name here violates the locality ideal, and also has a few other problems to consider:
- What happens if we rename this cog? Well, the method would break.
- What happens if you try to inherit from this cog, and then implement your own `_get_random_die` method? Then this would create surprising behavior in the inherited `roll` command, which would not use the new helper.

As a rule of thumb, always use `cls` or `self` when calling a method in your class, and don't stick your class name into `super()` either.

Reply:
Thanks for the explanation. I've been wondering what's considered best practice in such a case.